webapp/src/pages/profileList/components/ProfileListComponent.js

```diff
@@ -0,0 +1,91 @@
+import React, { Component } from "react";
+import { withRouter} from "react-router";
+ import ReactTable from 'react-table';
+import { profileService} from "../../../services/profilelist";
+
+const DEFAULT_EVENT_ID = process.env.REACT_APP_DEFAULT_EVENT_ID || 1;
+
+class ProfileListComponent extends Component{
+    constructor(props){
+        super(props);
+        this.state = {
+            List:[],
+            isEmpty:true,
+            loading:true,
+            error: ""
+        }
+}
+
+ isListEmpty(){
+
+    return this.state.List.length === 0;
+ }
+
+ componentDidMount(){ 
+    this.setState({List:profileService.getProfilesList(DEFAULT_EVENT_ID)
+                         .then(results => {
+                                this.setState(
+                                    {
+                                        List : results.List,
+                                        isEmpty:this.isListEmpty(),
```

Comment:
Rather pass results.List to your isListEmpty function, because at the moment when isListEmpty executes, the contents of state is unknown (see https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous)

Reply:
I understand. Sorted.